float8_experimental/float8_dynamic_linear.py

```diff
@@ -44,6 +54,17 @@ def backward(ctx, gradY):
         return fp8_tensor, None
 
 
+def cast_to_float8_e4m3fn(
```

Comment:
nit: put in utils.py?

Reply:
this is code motion right?

Reply:
This is moving from a method on the `Float8DynamicLinear` module to a standalone function so that it can be called from `fsdp_pre_all_gather()`.

Reply:
Since this is mainly code motion and is the cast specific to dynamic scaling, I think it might be better to leave it in this file.

Reply:
ok, maybe move below Float8DynamicLinear to keep the main class first?